torch/distributions/gamma.py

```diff
@@ -80,3 +80,9 @@ def _natural_params(self):
 
     def _log_normalizer(self, x, y):
         return torch.lgamma(x + 1) + (x + 1) * torch.log(-y.reciprocal())
+
+    def cdf(self, value):
+        value = torch.as_tensor(value, dtype=self.rate.dtype, device=self.rate.device)
```

Comment:
I believe in previous discussion `torch.distributions` maintainers had agreed to limit `torch.as_tensor()` to `.__init__()` methods, and not allow its use in other methods like `.log_prob()` and `.cdf()`.  If we were to allow `torch.as_tensor()` in those methods, that change would best be made in a single PR that changed all distribution methods to have comparable behavior.

I believe you can simply delete this line.

Reply:
Sounds good. Shall I drop [the same line](https://github.com/pytorch/pytorch/pull/72518/files#diff-abf0e6a8a075a1468e3b97d4245fadd3dfffca74f0a6dc3756534cbb3d83d663L66) from `log_prob` while I'm at it?

Reply:
hmm, while `.log_prob()` shouldn't use `torch.as_tensor()`, I'd leave it for now, avoiding risk of rollback of this PR in case something unrelated breaks. But up to you.